T4MVCExtensions/T4TypeHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace T4MVCExtensions
+{
+    public static class T4TypeHelper
+    {
+        public static bool IsT4MVC(this Type t)
+        {
+            if (t.FullName.StartsWith("T4MVC"))
+                return true;
+
+            var attribute = t.GetCustomAttributes(typeof(GeneratedCodeAttribute), false)
+                .OfType<GeneratedCodeAttribute>().FirstOrDefault();
+
+            if (attribute != null)
+            {
+                if (attribute.Tool == "T4MVC")
+                    return true;
+            }
+
+            if (t.IsNested)
```

Comment:
But there are nested inside the partial class and not the derived class. So I would expect there DeclaringType to be the partial class, and hence not return true to `IsT4MVC`. Maybe I'm confused :)